pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -458,14 +458,62 @@ func GetSecurityGroups(is *InstanceService, sg_param []openstackconfigv1.Securit
 	return sgIDs, nil
 }
 
+func trunkSupportNeeded(config *openstackconfigv1.OpenstackProviderSpec) bool {
+	if config.Trunk == true {
+		return true
+	}
+	for _, portCreateOpts := range config.Ports {
+		if *portCreateOpts.Trunk {
+			return true
+		}
+	}
+	return false
+}
+
+func getOrCreateTrunk(is *InstanceService, port *ports.Port, machineTags []string) (*trunks.Trunk, error) {
+	allPages, err := trunks.List(is.networkClient, trunks.ListOpts{
+		Name:   port.Name,
+		PortID: port.ID,
+	}).AllPages()
+	if err != nil {
+		return nil, fmt.Errorf("Searching for existing trunk for server err: %v", err)
```

Comment:
Please update the error message to include the port id, such as:
```suggestion
		return nil, fmt.Errorf("Searching for existing trunk for port %q err: %v", port.ID, err)
```